flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/batch/sql/SortTest.scala

```diff
@@ -46,6 +46,134 @@ class SortTest extends TableTestBase {
+    env.executeSql(s"""
+                      |CREATE TABLE Source (
+                      |  a int,
+                      |  b bigint,
+                      |  c string,
+                      |  d string,
+                      |  e string
+                      | ) WITH (
+                      |  'connector' = 'values',
+                      |  'bounded' = 'true'
+                      |)
+                      |""".stripMargin)
+    val query = "SELECT * FROM Source order by d"
+    val table = env.sqlQuery(query)
+    env.registerTable("SortedTable", table)
+    env.executeSql("""
+                     |CREATE TABLE sink1 (
+                     |   a int,
+                     |   b bigint,
+                     |   c string
+                     |) WITH (
+                     |   'connector' = 'filesystem',
+                     |   'format' = 'testcsv',
+                     |   'path' = '/tmp/test'
+                     |)
+                     |""".stripMargin)
+    env.executeSql("""
+                     |CREATE TABLE sink2 (
+                     |   a int,
+                     |   b bigint,
+                     |   c string,
+                     |   d string
+                     |) WITH (
+                     |   'connector' = 'filesystem',
+                     |   'format' = 'testcsv',
+                     |   'path' = '/tmp/test'
+                     |)
+                     |""".stripMargin)
+    val stmtSet = env.createStatementSet()
+    stmtSet.addInsertSql("INSERT INTO sink1 select a, b, e from SortedTable")
+    stmtSet.addInsertSql("INSERT INTO sink2 select a, b, d, e from SortedTable")
+
+    util.verifyExecPlan(stmtSet)
+  }
+
+  @Test
+  def testSortedViewResultIntoSingleSink(): Unit = {
+    val env = util.tableEnv
+    env.executeSql(s"""
+                      |CREATE TABLE Source (
+                      |  a int,
+                      |  b bigint,
+                      |  c string,
+                      |  d string,
+                      |  e string
+                      | ) WITH (
+                      |  'connector' = 'values',
+                      |  'bounded' = 'true'
+                      |)
+                      |""".stripMargin)
+    val query = "SELECT * FROM Source order by d"
+    val table = env.sqlQuery(query)
+    env.registerTable("SortedTable", table)
+    env.executeSql("""
+                     |CREATE TABLE sink2 (
+                     |   a int,
+                     |   b bigint,
+                     |   c string,
+                     |   d string
+                     |) WITH (
+                     |   'connector' = 'filesystem',
+                     |   'format' = 'testcsv',
+                     |   'path' = '/tmp/test'
+                     |)
+                     |""".stripMargin)
+    val stmtSet = env.createStatementSet()
+    stmtSet.addInsertSql("INSERT INTO sink2 select a, b, d, e from SortedTable")
+    util.verifyExecPlan(stmtSet)
```

Comment:
This test is not related to the bug. I think we can remove this.

Reply:
> This test is not related to the bug. I think we can remove this.

Done!